### PR TITLE
add nil check for TableConvertor in response.go

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**What this PR does / why we need it**:
check if scope.TableConvertor is nil,  else call scope.TableConvertor.ConvertToTable(ctx, result, opts) directly may panic
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
@xiangpengzhao   I run the error case in my local environment, and it passed, and this seems have no relation with my modification

**Comment by xiangpengzhao** [comment]
Seems like a flake. https://github.com/kubernetes/kubernetes/issues/50943

/retest

File: staging/src/k8s.io/apiserver/pkg/endpoints/handlers/response.go [hunk 1]
```diff
@@ -114,6 +114,11 @@ func transformResponseObject(ctx request.Context, scope RequestScope, req *http.
 				return
 			}
 
+			if scope.TableConvertor == nil {
+				err = newNotAcceptableError("scope.TableConvertor is nil")
```
Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/endpoints/handlers/response.go]: will this end up in the response? This should be an error understandable by the user, or http 500 if it's really internal.

Comment (hzxuzhonghu) [inline staging/src/k8s.io/apiserver/pkg/endpoints/handlers/response.go]: OK， I will return another err not 406 but 500

**Comment by hzxuzhonghu** [comment]
@sttts could you review again?

File: staging/src/k8s.io/apiserver/pkg/endpoints/handlers/response.go [hunk 2]
```diff
@@ -114,6 +114,11 @@ func transformResponseObject(ctx request.Context, scope RequestScope, req *http.
 				return
 			}
 
+			if scope.TableConvertor == nil {
```
Comment (smarterclayton) [inline staging/src/k8s.io/apiserver/pkg/endpoints/handlers/response.go]: We should never get into this code path - this is a programmer error, so we should treat it as such and ensure that it is not possible to enter this code path with a nil table convertor (by tightening earlier checks, not late checking).

Comment (hzxuzhonghu) [reply staging/src/k8s.io/apiserver/pkg/endpoints/handlers/response.go]: OK，earlier logic guarantee no nil.   I will close this.

**Pull request closed without merging**